services/QuillLMS/app/controllers/students_controller.rb

```diff
@@ -102,7 +101,13 @@ def join_classroom
   private def redirect_to_profile
     @current_user = current_user
     classroom_id = params["classroom"]
-    if classroom_id && (Classroom.find_by(id: classroom_id).nil? || StudentsClassrooms.find_by(student_id: @current_user.id, classroom_id: classroom_id).nil?)
+    unit_id = params["unit_id"]
+
+    return redirect_to profile_path if current_user && !current_user.student?
```

Comment:
You could combine Ruby's built-in nil-checking with `unless` here: `return redirect_to profile_path unless current_user&.student?`

(`current_user&.student?` will return `nil` if `current_user` is `nil`, and it won't try to check the value of `student?`)

Reply:
good call-- I always forget that ruby has that `&.` feature